lua/pears.lua

```diff
@@ -172,7 +172,8 @@ function M._handle_return(bufnr)
         direction = Pointer.Direction.Forward
       }
 
-      if after and after.leaf and after.leaf.key == before.leaf.key then
+      if (after and after.leaf and after.leaf.key == before.leaf.key) or
+        (not after and input:expand(nil, true)) then
```

Comment:
I'm not a fan of doing this call in the condition since it contains side effects. I think we should break it out into another conditional to make it easier to follow.

Reply:
Agreed.